### PR TITLE
Feature/set locale

### PR DESCRIPTION
**Pull request by fitja**
## Proposed Changes

Currently the `locale` parameter is sent along when calling `start()`, but there is no way to override it.

This PR introduces `setLocale` API which can be called before or after `start()`.
The Leanplum class now have private property for overriding the system locale.

Similar approach has been done in https://github.com/Leanplum/Leanplum-JavaScript-SDK/pull/140

### CODE REVIEW

File: LeanplumSDKApp/LeanplumSDKTests/Classes/LeanplumTest.m [hunk 1]
```diff
@@ -1075,6 +1075,45 @@ - (void) test_set_device_id_after_start
     XCTAssertTrue([[LPAPIConfig sharedConfig].deviceId isEqualToString:deviceId]);
 }
 
+/**
+ * Tests setting the user locale after Leanplum start
+ */
+- (void) test_set_locale
```
Comment (nzagorchev) [inline LeanplumSDKApp/LeanplumSDKTests/Classes/LeanplumTest.m:1081]: You can simplify the test to:
```objc
- (void) test_set_locale
{
    NSLocale *locale = [[NSLocale alloc] initWithLocaleIdentifier:@"ab_CD"];
    
    XCTestExpectation *request_expectation =
    [self expectationWithDescription:@"request_expectation"];
    [LPRequestSender validate_request_args_dictionary:^(NSDictionary *args) {
        XCTAssertTrue([args[LP_KEY_LOCALE] isEqualToString:[locale localeIdentifier]]);
        XCTAssertFalse([args[LP_KEY_LOCALE] isEqualToString:[[NSLocale currentLocale] localeIdentifier]]);
        [request_expectation fulfill];
    }];
    
    [Leanplum setLocale:locale];
    [self waitForExpectations:@[request_expectation] timeout:1.0];
}
```

File: LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m [hunk 2]
```diff
@@ -2249,6 +2249,14 @@ + (void)setTrafficSourceInfoInternal:(NSDictionary *)info
     [[LPRequestSender sharedInstance] send:request];
 }
 
++ (void)setLocale:(NSLocale *)locale
+{
```
Comment (nzagorchev) [inline LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m:2253]: What is the use case - would this be called before starting Leanplum or at some point after it started?
If it has started, the locale in the start API call will be the system one.

**Review by nzagorchev — CHANGES_REQUESTED** [review]

File: LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m (hunk 2 above)
Comment (fitja) [reply LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m:2253]: The idea was for calling it after Leanplum is started, in order to simplify a bit complex variations of starting.

If you think it is better to move it to start, and/or to add some stored property for overriding the system's locale, so the start would take that one, I'm ok with it as well.

**Review by nzagorchev — APPROVED** [review]

**Comment by batadamnjanovic** [comment]
Hi @nzagorchev, thanks for the collaboration.

Do you maybe know when will be the release with this change? It will mean a lot to us to organise some things in this sprint.

Thanks!

**Comment by nzagorchev** [comment]
Hi @batadamnjanovic,
We plan a release in early January.

Can you update the PR, so I can merge it into master after that?
Thanks

File: LeanplumSDKApp/LeanplumSDKTests/Classes/LeanplumTest.m (hunk 1 above)
Comment (fitja) [reply LeanplumSDKApp/LeanplumSDKTests/Classes/LeanplumTest.m]: Tests has been updated

File: LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m (hunk 2 above)
Comment (fitja) [reply LeanplumSDK/LeanplumSDK/Classes/Internal/Leanplum.m:2269]: Improved in a way that `setLocale` now can be called both before and after Leanplum is started

**Comment by fitja** [comment]
Hi @nzagorchev the PR has been updated (proposed changes) also the branch has been updated with `master`
Can you check if there is anything more to update/change so the PR can be merged?
Thanks!

**Pull request merged**